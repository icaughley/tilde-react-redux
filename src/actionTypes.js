export const ERROR_SUFFIX = '_ERROR';

export const GET_PROJECTS = 'GET_PROJECTS';
export const CREATE_PROJECT = 'CREATE_PROJECT';
export const UPDATE_PROJECT = 'UPDATE_PROJECT';
export const DELETE_PROJECT = 'DELETE_PROJECT';
export const SET_PROJECT_CLOAKED = 'SET_PROJECT_CLOAKED';
export const GET_INVOICING = 'GET_INVOICING';
export const GET_BILLABLE_PROJECTS = 'GET_INVOICING_PROJECTS';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const OPEN_MESSAGE = 'OPEN_MESSAGE';
export const CLOSE_MESSAGE = 'CLOSE_MESSAGE';
export const SET_SELECTED_PROJECT = 'SET_SELECTED_PROJECT';
export const GET_WORK = 'GET_WORK';
