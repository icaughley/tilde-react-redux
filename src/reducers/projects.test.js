import deepFreeze from "deep-freeze";
import reduce from "./projects";
import {CREATE_PROJECT, DELETE_PROJECT, GET_PROJECTS, SET_PROJECT_CLOAKED, UPDATE_PROJECT} from "../actionTypes";

describe('Projects reducer', () => {
    it('provides the initial state ', () => {
        expect(reduce()).toEqual({});
    });

    it('ignores unrecognised actions', () => {
        expect(reduce({}, {type: 'UNRECOGNISED'})).toEqual({});
    });

    it('handles action of type GET_PROJECTS', () => {
        const data1 = {id: 12, name: "a"};
        const data2 = {id: 13, name: "b"};
        const action = {type: GET_PROJECTS, payload: {data: [data1, data2]}};
        const state = {};

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action)).toMatchSnapshot();
    });

    it('handles action of type SET_PROJECT_CLOAKED', () => {
        const action = {type: SET_PROJECT_CLOAKED, payload: {data: {id: 12, cloaked: true}}};
        const state = {12: {id: 12, cloaked: false}};

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action)).toMatchSnapshot();
    });

    it('handles action of type UPDATE_PROJECT', () => {
        const action = {type: UPDATE_PROJECT, payload: {data: {id: 13, name: "Bob"}}};
        const state = {13: {id: 13, name: "Foo"}};

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action)).toMatchSnapshot();
    });

    it('handles action of type CREATE_PROJECT', () => {
        const action = {type: CREATE_PROJECT, payload: {data: {id: 14, name: "Bob"}}};
        const state = {};

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action)).toMatchSnapshot();
    });

    it('handles action of type DELETE_PROJECT', () => {
        const action = {type: DELETE_PROJECT, payload: {data: {id: 15}}};
        const state = {15: {id: 15, name: "bob"}};

        deepFreeze(state);
        deepFreeze(action);

        expect(reduce(state, action)).toEqual({});
    });
});
