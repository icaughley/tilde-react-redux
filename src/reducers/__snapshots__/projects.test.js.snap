exports[`Projects reducer handles action of type CREATE_PROJECT 1`] = `
Object {
  "14": Object {
    "id": 14,
    "name": "Bob",
  },
}
`;

exports[`Projects reducer handles action of type GET_PROJECTS 1`] = `
Object {
  "12": Object {
    "id": 12,
    "name": "a",
  },
  "13": Object {
    "id": 13,
    "name": "b",
  },
}
`;

exports[`Projects reducer handles action of type SET_PROJECT_CLOAKED 1`] = `
Object {
  "12": Object {
    "cloaked": true,
    "id": 12,
  },
}
`;

exports[`Projects reducer handles action of type UPDATE_PROJECT 1`] = `
Object {
  "13": Object {
    "id": 13,
    "name": "Bob",
  },
}
`;
